### PR TITLE
[SR-7612] Add host target triple for arm64/aarch64 machines.

### PR DESCRIPTION
**Pull request by futurejones**
Added host target triple for arm64/aarch64.
Fixed incorrect armv7 triple.
Refactored bootstrap "compute the build paths"
Refactored Triple.swift "os and machine" dif's

New OS's and processor/machine types are currently being added in an addhoc manner. Refactoring the Utilities/bootstrap and Triple.swift addresses [[SR-7612]](https://bugs.swift.org/browse/SR-7612). While this solution is still hardcoded it provides conformity between machines and a consistent naming type for triples. 
i.e [machine][OS] - x86Linux, arm64Linux.
This will provide an easy path for adding new machines and OS's in the future.

### CODE REVIEW

**Comment by aciidb0mb3r** [comment]
@swift-ci please smoke test

File: Utilities/bootstrap [hunk 1]
```diff
@@ -895,14 +895,21 @@ def main():
...
-    elif platform.system() == 'Linux' and platform.machine() == 'arm':
-        build_target = 'arm-linux-gnueabihf'
+    elif platform.system() == 'Linux':              
+        if platform.machine() == 'x86_64':
+            build_target = "x86_64-unknown-linux"
+        elif platform.machine() == 's390x':
+            build_target = "s390x-unknown-linux"
+        elif platform.machine() == 'ppc64le':
+            build_target = 'powerpc64le-unknown-linux'
+        elif platform.machine().startswith("armv7"):
```
Comment (aciidb0mb3r) [inline Utilities/bootstrap]: Is this correct? The previous code was checking this: `platform.machine() == 'arm'`

File: Utilities/bootstrap [hunk 2]
```diff
@@ -895,14 +895,21 @@ def main():
...
+            build_target = 'armv7-unknown-linux-gnueabihf'
```
Comment (aciidb0mb3r) [inline Utilities/bootstrap]: Similarly, the previous code has this target triple: `arm-linux-gnueabihf`

File: Utilities/bootstrap (hunk 1 above)
Comment (futurejones) [reply Utilities/bootstrap]: Yes, 
platform.machine() == 'arm' will fail on some machines.
e.g. RaspberryPi3 platform.machine() returns 'armv7l'
By using .startswith("armv7") we can identify the arm machines that are compatible with the 'armv7-unknown-linux-gnueabihf' triple. 
.startswith() has also been included in https://github.com/uraimo/buildSwiftOnARM/blob/master/swiftpm.diffs/1546.diff

File: Utilities/bootstrap (hunk 2 above)
Comment (futurejones) [reply Utilities/bootstrap]: Yes, 
This change has also been included in https://github.com/uraimo/buildSwiftOnARM/blob/master/swiftpm.diffs/1546.diff
It has been tested on a variety of armv7 boards such as RaspberryPi, Asus Tinkerboard, ODROID Xu4.

**Comment by aciidb0mb3r** [comment]
Great, thanks!

**Comment by futurejones** [comment]
Is it possible to add this PR to the "swift-4.1-branch" ?

**Comment by aciidb0mb3r** [comment]
@futurejones Nope, Swift 4.1 is already released! https://swift.org/blog/swift-4-1-released/

**Pull request merged**